src/gpu/intel/ocl/rnn/rnn_grid.cl

```diff
@@ -1005,28 +1005,57 @@ void cell_common_inner(const_wei_layer_cell_t wei_layer,
-                        gate_idx++) {
-                    G[gate_idx] = C[gate_idx][n_l][c_l];
-                    B[gate_idx] = convert_float(
-                            ctx.lstm.bias[off_ker_bias(dims.dhc, gate_idx, c)]);
-                }
-                vanilla_lstm_gates_t g
-                        = vanilla_lstm_compute_gates(G, B, ctx.lstm.tm_scales);
-                vanilla_lstm_store(gates.ptr, gates.strides.mb, states.ptr,
-                        ctx.lstm.c_states, ctx.lstm.c_states_iter,
-                        states.strides.mb, dims.dhc, n, c, ctx.lstm.tm_cscale,
-                        g);
-            } else if (CELL_KIND == VANILLA_RNN) {
-                float g = vanilla_rnn_compute_gates(C[0][n_l][c_l],
-                        ctx.rnn.bias[off_ker_bias(dims.dhc, 0, c)],
-                        ctx.rnn.alpha, ctx.rnn.tm_scales);
-                store_vanilla_rnn(gates.ptr, gates.strides.mb, states.ptr,
-                        states.strides.mb, dims.dhc, n, c, g);
+#if CELL_KIND == VANILLA_LSTM
+            float G[vanilla_lstm_n_gates];
+            float B[vanilla_lstm_n_bias];
+            for (int gate_idx = 0; gate_idx < vanilla_lstm_n_gates;
+                    gate_idx++) {
+                G[gate_idx] = C[gate_idx][n_l][c_l];
+                B[gate_idx] = convert_float(
+                        ctx.lstm.bias[off_ker_bias(dims.dhc, gate_idx, c)]);
+            }
+            vanilla_lstm_gates_t g
+                    = vanilla_lstm_compute_gates(G, B, ctx.lstm.tm_scales);
+            vanilla_lstm_store(gates.ptr, gates.strides.mb, states.ptr,
+                    ctx.lstm.c_states, ctx.lstm.c_states_iter,
+                    states.strides.mb, dims.dhc, n, c, ctx.lstm.tm_cscale, g);
+#elif CELL_KIND == VANILLA_RNN
+            float g = vanilla_rnn_compute_gates(C[0][n_l][c_l],
+                    ctx.rnn.bias[off_ker_bias(dims.dhc, 0, c)], ctx.rnn.alpha,
+                    ctx.rnn.tm_scales);
+            store_vanilla_rnn(gates.ptr, gates.strides.mb, states.ptr,
+                    states.strides.mb, dims.dhc, n, c, g);
+#elif CELL_KIND == LBR_GRU
+            //__global AUX_DATA_T *scratch_cell -> ctx.lbr_gru.scratch_cell
+            //__global WS_STATE_DATA_T *src_iter -> ctx.lbr_gru.hidden_states_iter
+            //__global ... *h_states_t_l -> states.ptr
+            lbr_gru_gates_t g = compute_gates_lbr_gru(scratch_gates.ptr,
+                    ctx.lbr_gru.scratch_cell, ctx.lbr_gru.bias,
+                    ctx.lbr_gru.tm_scales, scratch_gates.strides.mb, dims.dhc,
+                    n, c);
+            float Wh_b = g.Wh_b;
+            float G0 = g.G[0];
+            float G1 = g.G[1];
+            float G2 = g.G[2];
+            float Ht = G0
+                            * TO_REF(
+                                    ctx.lbr_gru.hidden_state_iter[cell_ws_state(
+                                            states.strides.mb, n, c)])
+                    + (1 - G0) * G2;
```

Comment:
Please wrap the `Ht` calculation into `compute_gates_lbr_gru`., we can thing just get rid of all these intermediate variables.

Additionally, please update `elemwise_fwd` to use  `compute_gates_lbr_gru` to avoid code duplication.